### DISCUSSION TITLE
Commit by Unknown User in a Private Repository

### DISCUSSION THREAD

**Discussion by TestPrab** [community/community#50820] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Bug

### Body

There's a private repository for which I have the access, while committing in that repository during the 
`git config -- local user.email  <your email > `
I gave someone else email by mistake as mine and theirs's are almost same , and for 
`git config --local user.name <your username>` 
I gave my username. 

After I pushed my code into the branch, I saw that the  username  by which commit was made , was of someone else and it did not belong to me . 
But the commit message was the same with which I committed along with the files that were changed in the commit. 

This unknown person did not had access to the repository and also I checked the participants of the repository after the commit if they got the access , they were  not present in the participants list as well . 

I would like to know now if the person has the access to the code base of the repository ? 
(As currently I cannot see the username in the list of participants for the particular repository ) 
And what steps to follow to remove this commit ? 
(Shall I delete the entire branch? Create a new branch and commit again with the corrected config?  Or how it should be)

Thanks For the help

**Comment by airtower-luna** [comment, marked as answer]
GitHub uses *only* the author email recorded in the commit (that's what you set with `user.email`) to attribute it to a user. That doesn't do anything else, the user won't have access to your repository.

If you want to fix the author email (so the commits correctly show up as yours) you can [rewrite the repository history](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History), but mind that anyone who already has the current history in their local repository will have to manually accept the changed history.

**Reply by TestPrab** [reply]
Thanks for the help

**Reply by airtower-luna** [reply]
Glad I could help! Please don't forget to mark the answer. :slightly_smiling_face:
